### PR TITLE
CIRCSTORE-539: added new request.item fields and db indexing into request schema

### PR DESCRIPTION
**Pull request by kapil-epam**
https://folio-org.atlassian.net/browse/CIRCSTORE-539

### CODE REVIEW

File: ramls/request.json [hunk 1]
```diff
@@ -133,6 +133,22 @@
         "barcode": {
           "description": "barcode of the item",
           "type": "string"
+        },
+        "itemEffectiveLocationId": {
+          "description": "Item's effective location",
+          "type": "string"
```
Comment (gurleenkaurbp) [inline ramls/request.json:139]: Should there be a pattern or $ref attributes for the ID's ? for e.g. "$ref": "raml-util/schemas/uuid.schema"

File: ramls/request.json [hunk 2]
```diff
@@ -133,6 +133,22 @@
...
+        },
+        "itemEffectiveLocationName": {
+          "description": "Item's effective location name",
+          "type": "string"
+        },
+        "retrievalServicePointId": {
+          "description": "Item's location primary service point",
+          "type": "string"
```
Comment (gurleenkaurbp) [inline ramls/request.json:147]: Should there be a pattern or $ref attributes for the ID's ? for e.g. "$ref": "raml-util/schemas/uuid.schema"

**Review by gurleenkaurbp — COMMENTED** [review]
Considering that new fields are getting added in request should the interface  "id": "request-storage",
      "version": "6.1", be bumped up ? 
      Are these fields mandatory or optional ?

File: ramls/request.json (hunk 1 above)
Comment (kapil-epam) [reply ramls/request.json]: fixed:=> https://github.com/folio-org/mod-circulation-storage/pull/498/commits/9a80571bd56b1a2d0f308c1bfb41dffdaf162896

File: ramls/request.json (hunk 2 above)
Comment (kapil-epam) [reply ramls/request.json]: fixed:=> https://github.com/folio-org/mod-circulation-storage/pull/498/commits/9a80571bd56b1a2d0f308c1bfb41dffdaf162896

**Review by gurleenkaurbp — APPROVED** [review]

**Pull request merged**